app/mailers/candidate_mailer.rb

```diff
@@ -250,6 +250,66 @@ def reference_received(reference)
     )
   end
 
+  def offer_10_day(application_choice)
+    @first_name = application_choice.application_form.first_name
+    @course_name = application_choice.current_course_option.course.name
+    @candidate = application_choice.application_form.candidate
+    @provider_name = application_choice.current_course_option.provider.name
+
+    email_for_candidate(
+      application_choice.application_form,
+      subject: I18n.t!('candidate_mailer.offer_day.subject', provider_name: @provider_name),
+    )
+  end
+
+  def offer_20_day(application_choice)
+    @first_name = application_choice.application_form.first_name
+    @course_name = application_choice.current_course_option.course.name
+    @candidate = application_choice.application_form.candidate
+    @provider_name = application_choice.current_course_option.provider.name
+
+    email_for_candidate(
+      application_choice.application_form,
+      subject: I18n.t!('candidate_mailer.offer_day.subject', provider_name: @provider_name),
+    )
+  end
+
+  def offer_30_day(application_choice)
+    @first_name = application_choice.application_form.first_name
+    @course_name = application_choice.current_course_option.course.name
+    @candidate = application_choice.application_form.candidate
+    @provider_name = application_choice.current_course_option.provider.name
+
+    email_for_candidate(
+      application_choice.application_form,
+      subject: I18n.t!('candidate_mailer.offer_day.subject', provider_name: @provider_name),
+    )
+  end
+
+  def offer_40_day(application_choice)
+    @first_name = application_choice.application_form.first_name
+    @course_name = application_choice.current_course_option.course.name
+    @candidate = application_choice.application_form.candidate
+    @provider_name = application_choice.current_course_option.provider.name
+
+    email_for_candidate(
+      application_choice.application_form,
+      subject: I18n.t!('candidate_mailer.offer_day.subject', provider_name: @provider_name),
+    )
+  end
+
+  def offer_50_day(application_choice)
+    @first_name = application_choice.application_form.first_name
+    @course_name = application_choice.current_course_option.course.name
+    @candidate = application_choice.application_form.candidate
+    @provider_name = application_choice.current_course_option.provider.name
+
+    email_for_candidate(
+      application_choice.application_form,
+      subject: I18n.t!('candidate_mailer.offer_50_day.subject', provider_name: @provider_name),
+    )
+  end
```

Comment:
Can these be one method with a `days` argument?

Reply:
I think that introduces too much logic to the yml